lib/puppet/provider/group/windows_adsi.rb

```diff
@@ -39,6 +39,11 @@ def members_to_s(users)
     return '' if users.nil? or !users.kind_of?(Array)
     users = users.map do |user_name|
       sid = Puppet::Util::Windows::SID.name_to_sid_object(user_name)
+      if !sid
+        resource.debug("#{user_name} (unresolvable to SID)")
+        next user_name
```

Comment:
Weird, I called out the same section, but for other reasons... https://github.com/puppetlabs/puppet/pull/4024#discussion_r32166350